backend/elo.py

```
@@ -0,0 +1,109 @@
+        timestamp: ISO 8601 timestamp to record on both entries.
+
+    No-op when winner == loser. Self-matches would alias `w` and `l` to
+    the same dict and double-increment `games`, silently corrupting state.
+    Stray self-pairs from upstream parsing should be ignored, not recorded.
+    """
+    if winner == loser:
+        return
+    w = ratings.setdefault(
+        winner, {"rating": INITIAL_RATING, "games": 0, "last_updated": timestamp}
+    )
+    l = ratings.setdefault(
```

Ruff (0.15.11)

[error] Ambiguous variable name: `l`

(E741)